components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -269,6 +270,167 @@ public Response getAPI(String apiId, String xWSO2Tenant, String ifNoneMatch,
+     */
+    @Override
+    public Response deleteApiEndpoint(String apiId, String endpointUuid, MessageContext messageContext)
+            throws APIManagementException {
+        try {
+            APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
+            //validate if api exists
+            CommonUtils.validateAPIExistence(apiId);
+            //validate API Endpoint
+            APIEndpointInfo existingApiEndpoint = apiProvider.getAPIEndpointByUUID(apiId, endpointUuid);
+            if (existingApiEndpoint != null) {
+                apiProvider.deleteAPIEndpointById(endpointUuid);
+                if (log.isDebugEnabled()) {
+                    log.debug("The API endpoint " + endpointUuid + " has been deleted from the the API " + apiId);
+                }
+                return Response.ok().build();
+            } else {
+                throw new APIMgtResourceNotFoundException("Couldn't retrieve an existing API Endpoint with ID: "
+                        + endpointUuid + " for API with UUID " + apiId,
+                        ExceptionCodes.from(ExceptionCodes.API_ENDPOINT_NOT_FOUND, endpointUuid));
+            }
+        } catch (APIManagementException e) {
+            if (RestApiUtil.isDueToResourceNotFound(e) || RestApiUtil.isDueToAuthorizationFailure(e)) {
+                RestApiUtil.handleResourceNotFoundError(RestApiConstants.RESOURCE_PATH_API_ENDPOINTS,
+                        endpointUuid, e, log);
+            } else {
+                String errorMessage = "Error while deleting the API specific API endpoint with ID :" +
+                        endpointUuid + " for API " + apiId + " " + e.getMessage();
+                RestApiUtil.handleInternalServerError(errorMessage, e, log);
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public Response updateApiEndpoint(String apiId, String endpointId, APIEndpointDTO apIEndpointDTO,
+            MessageContext messageContext) throws APIManagementException {
+        try {
+            APIRevision apiRevision = ApiMgtDAO.getInstance().checkAPIUUIDIsARevisionUUID(apiId);
+            if (apiRevision != null && apiRevision.getApiUUID() != null) {
+                throw new APIManagementException("Cannot Update API Endpoint in Revision View : " + endpointId,
+                        ExceptionCodes.ERROR_UPDATING_API_ENDPOINT_API);
+            }
+            APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
+            String organization = RestApiUtil.getValidatedOrganization(messageContext);
+            //validate if api exists
+            CommonUtils.validateAPIExistence(apiId);
+            PublisherCommonUtils.updateAPIEndpoint(apiId, endpointId, apIEndpointDTO, organization, apiProvider);
+            APIEndpointInfo updatedAPIEndpoint = apiProvider.getAPIEndpointByUUID(apiId, endpointId);
+            APIEndpointDTO updatedAPIEndpointDTO = APIMappingUtil.fromAPIEndpointToDTO(updatedAPIEndpoint);
```

Comment:
I couldn't find any logic to handling invalid endpoint id passed here. 404 error should be sent if endpoint is not found for the given id.